core/src/main/scala/org/apache/spark/MapOutputTracker.scala

```diff
@@ -161,7 +162,7 @@ private class ShuffleStatus(numPartitions: Int) extends Logging {
    */
   def removeOutputsOnHost(host: String): Unit = withWriteLock {
     logDebug(s"Removing outputs for host ${host}")
-    removeOutputsByFilter(x => x.host == host)
+    removeOutputsByFilter(x => x.asInstanceOf[BlockManagerId].host == host)
```

Comment:
This is making assumption about the type - while it would work currently with existing implementation, how is this going to work going forward ?

(Same comment for all other methods where `asInstanceOf[BlockManagerId]` is being done).